src/presetsController.ts

```diff
@@ -1016,16 +1022,25 @@ export class PresetsController {
         const validator = await loadSchema(schemaFile);
         const is_valid = validator(presetsFile);
         if (!is_valid) {
-            const errors = validator.errors!;
-            log.error(localize('invalid.file.error', 'Invalid kit contents {0} ({1}):', path.basename(file), file));
-            for (const err of errors) {
-                if (err.params && 'additionalProperty' in err.params) {
-                    log.error(` >> ${err.dataPath}: ${err.message}: ${err.params.additionalProperty}`);
-                } else {
-                    log.error(` >> ${err.dataPath}: ${err.message}`);
+            const showErrors = (logFunc: (...x: any[]) => void) => {
+                const errors = validator.errors!;
+                logFunc(localize('unsupported.presets', 'Unsupported presets detected in {0}. Support is limited to features defined by version {1}.', file, maxSupportedVersion));
+                for (const err of errors) {
+                    if (err.params && 'additionalProperty' in err.params) {
+                        logFunc(` >> ${err.dataPath}: ${err.message}: ${err.params.additionalProperty}`);
+                    } else {
+                        logFunc(` >> ${err.dataPath}: ${err.message}`);
+                    }
                 }
+            };
+            if (validationErrorsAreWarnings) {
+                showErrors(log.warning);
+                return presetsFile;
+            } else {
+                showErrors(log.error);
+                log.error(localize('unsupported.presets.disable', 'If desired, unknown properties and macros can be ignored by using the {0} setting.', "'cmake.allowUnsupportedPresetsVersions'"));
```

Comment:
Maybe we can write something easier to understand, I am ok with the current explanation though.
localize('disable.unsupported.presets', 'In order to ignore the unknown properties and macros, modify the {0} setting.', "'cmake.allowUnsupportedPresetsVersions'")

Reply:
I did it. Thanks